tools/odbc/odbc_fetch.cpp

```diff
@@ -12,13 +13,185 @@ using duckdb::OdbcHandleStmt;
 OdbcFetch::~OdbcFetch() {
 }
 
-SQLRETURN OdbcFetch::Fetch(SQLHSTMT statement_handle, OdbcHandleStmt *stmt) {
+void OdbcFetch::IncreaseRowCount() {
+	if ((chunk_row + rowset_size) < current_chunk->size()) {
+		row_count += rowset_size;
+	} else {
+		row_count += current_chunk->size() - chunk_row;
+	}
+}
+
+SQLRETURN OdbcFetch::FetchNext(OdbcHandleStmt *stmt) {
+	try {
+		auto chunk = stmt->res->Fetch();
+		if (!chunk) {
+			resultset_end = true;
+			return SQL_NO_DATA;
+		}
+		if (!chunks.empty()) {
+			// only push the prior chunk because the new one is already "popped" to the current chunk
+			stack_prior_chunks.push(chunks.back().get());
+		}
+		chunks.emplace_back(move(chunk));
```

Comment:
This seems to me like the chunks are always fully materialized in the ODBC result. This should not be necessary for a cursor that does not allow scrolling backwards (e.g. `SQL_CURSOR_FORWARD_ONLY`), correct?

Since that seems like a common case and we don't want to require clients to always materialize the entire result in memory I think it is worth having a special case here where chunks are only materialized for cursors that allow scrolling backwards.

Reply:
That makes sense. Improvements were applied accordingly.